### ISSUE TITLE
VirtualServerRoute - Remote Location JWKs - Cache directory not created for jwks_uri causing nginx to fail.

### ISSUE THREAD

**Issue by uniquemind27** [nginxinc/kubernetes-ingress#3964]
I am trying to have JWT auth enabled for Nginx Ingress via VirtualServerRoute. I created a policy and applied the same against the VirtualServerRoute. It works fine when the path is /app1. But when the path is /iwa/app1 then I can see the config (`auth_jwt_key_request /iwa/app1_jwks_uri;`) which is expecting a dir called temp to be present under /var/cache/nginx. 

Config:
`proxy_cache_path /var/cache/nginx/iwa/app1_jwks_uri levels=1 keys_zone=/iwa/app1_jwks_uri:1m max_size=10m;`

Error message:
`[emerg] 23#23: mkdir() "/var/cache/nginx/iwa/app1_jwks_uri" failed (2: No such file or directory)`

Install the Nginx Ingres Controller. 
Apply the service, deployment and servicedefaults.
Apply the policies, VirtualServer & VirtualServerRoute.
You will notice the VirtualServer and VirtualServerRoute in invalid state. 

Error can be seen inside the nginx ingress contoller pod.
```
nginx@nginx-ingress-58f8fbbb46-kcshj:/$ nginx -t
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: [emerg] mkdir() "/var/cache/nginx/iwa/app1_jwks_uri" failed (2: No such file or directory)
nginx: configuration file /etc/nginx/nginx.conf test failed
```

Expected behavior:
Once the policy is applied successfully to the VirtualServerRoute I am expecting the nginx config to have a valid cache path so that nginx can load successfully. 

Versions:
Nginx Ingres Controller Version - 3.0.2
Helm chart version - 0.16.2
Nginx Plus 

Kubernetes: AKS
Client Version: v1.25.4
Kustomize Version: v4.5.7
Server Version: v1.26.0
[NginxIngress.tgz](https://github.com/nginxinc/kubernetes-ingress/files/11666214/NginxIngress.tgz)

**Comment by jasonwilliams14** [comment]
@uniquemind27 Hello there.
This particular issue was addressed in our 3.1.x release.
If you upgrade to our latest version, this issue should be resolved.
Hope that helps

**Comment by uniquemind27** [comment]
Thanks @jasonwilliams14 .

That did work for us after uplifting from 3.0.2 to 3.1.1.
